keep/providers/cloudwatch_provider/cloudwatch_provider.py

```diff
@@ -359,17 +364,20 @@ def _query(
         while response is None or response["status"] == "Running":
             self.logger.debug("Waiting for AWS cloudwatch query to complete...")
             time.sleep(1)
-            try:
-                response = logs_client.get_query_results(queryId=query_id)
-            except Exception:
-                # probably no permissions
-                self.logger.exception(
-                    "Error getting AWS cloudwatch query results - add logs:GetQueryResults permissions",
-                    extra={"kwargs": kwargs},
-                )
-                raise
-
-        results = response.get("results")
+            response = logs_client.get_query_results(queryId=query_id)
+            # Response in format List[{field: fieldName, value: fieldValue}]
+            # We need to convert it to List[Dict[fieldName: fieldValue]]
+            results = []
+            for result in response.get("results", []):
+                results.append({field["field"]: field["value"] for field in result})
+                # Trying to parse JSON of each field["value"]
+                for field in results[-1]:
+                    try:
+                        results[-1][field] = json.loads(results[-1][field])
+                    except json.JSONDecodeError:
+                        pass
+                if remove_ptr_from_results:
+                    results[-1].pop("@ptr", None)
```

Comment:
just making it a bit more straightforward 🙏🏼 
```suggestion
                key_value_dict = {}
                for field in result:
                    if remove_ptr_from_results and field.get("field") == "@ptr":
                        continue
                    value = field.get("value")
                    try:
                        value = json.loads(value)
                    except json.JSONDecodeError:
                        pass
                    key_value_dict[field.get("field")] = value
                results.append(key_value_dict)
```